blobnet/examples/speed.rs

```diff
@@ -9,74 +9,200 @@ struct Args {
+        let (blobnet, cache, tmpdir): (Arc<dyn Provider>, _, _) =
+            match (args.cache, args.cache_dir.clone()) {
+                (false, None) => (Arc::new(remote), None, None),
+                (true, _) => {
+                    let dir = tempdir().unwrap();
+                    let cached = Arc::new(Cached::new(remote, &dir, cache_page_size_bytes));
+                    (cached.clone(), Some(cached), Some(dir))
+                }
+                (_, Some(dir)) => {
+                    let cached = Arc::new(Cached::new(remote, dir, cache_page_size_bytes));
+                    (cached.clone(), Some(cached), None)
+                }
+            };
+
         let start = Instant::now();
-        let m = n / args.concurrency;
-        let mut set = JoinSet::new();
-        for i in 0..args.concurrency {
-            let c = client.clone();
-            let h = hash.clone();
-            set.spawn(async move {
-                let r = Some((i * m, ((i + 1) * m)));
-                let r = c.get(&h, r).await.unwrap();
-                drain(r).await.unwrap()
-            });
-        }
-        while let Some(res) = set.join_next().await {
-            let pn: u64 = res?;
-            assert_eq!(pn, m);
+        let n = fetch(
+            &blobnet,
+            &hash,
+            args.file_size_bytes as u64,
+            args.concurrency,
+            fetch_page_size_bytes,
+        )
+        .await?;
+        let latency = start.elapsed();
+        let thpt = throughput_mibps(n, latency);
+        times.push(latency);
+
+        println!();
+        println!("iteration {}", i);
+        println!("lat = {} us", latency.as_micros());
+        println!("thpt = {:.2} Gbit/s ({:.2} GB/s)", thpt, thpt / 8f64);
+
+        if let Some(cache) = cache {
+            let stats = cache.stats();
+            let pending_disk_write_pages_mb =
+                stats.pending_disk_write_bytes as f64 / (1 << 20) as f64;
+            if stats.pending_disk_write_pages > 0 {
+                print!(
+                    "waiting for {} ({} MB) pending cache writes..",
+                    stats.pending_disk_write_pages, pending_disk_write_pages_mb
+                );
+                for j in 0.. {
+                    if cache.stats().pending_disk_write_pages == 0 {
+                        break;
+                    }
+                    sleep(Duration::from_millis(10)).await;
+                    if j % 100 == 0 {
+                        print!(".");
+                        io::stdout().flush().unwrap();
+                    }
+                }
+            }
+            println!("done");
+
+            if let Some(tmpdir) = tmpdir {
+                tmpdir.close()?;
+            }
+
+            let latency = start.elapsed();
+            let thpt = throughput_mibps(n, latency);
+            println!(
+                "disk write thpt = {:.2} Gbit/s ({:.2} GB/s)",
+                thpt,
+                thpt / 8f64
```

Comment:
```suggestion
                thpt / 8_f64
```

nit: Didn't know skipping the `_` was allowed.